File: stream/src/main/java/bsky4j/stream/internal/atproto/_ATProtocolStream.java
package kbsky.stream.internal.atproto;

import kbsky.ATProtocol;
import kbsky.internal._ATProtocol;
import kbsky.stream.ATProtocolStream;
import kbsky.stream.api.atproto.SyncResource;

public class _ATProtocolStream implements ATProtocolStream {


    private final ATProtocol atproto;

    protected final SyncResource sync;

    public _ATProtocolStream(String uri) {
        this.atproto = new _ATProtocol(uri);
        this.sync = new _SyncResource(this.atproto, uri);
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public SyncResource sync() {
        return sync;
    }

}

File: stream/src/commonMain/kotlin/work/socialhub/kbsky/stream/internal/atproto/_ATProtocolStream.kt
package work.socialhub.kbsky.stream.internal.atproto

import work.socialhub.kbsky.stream.api.atproto.SyncResource
import work.socialhub.kbsky.ATProtocol
import work.socialhub.kbsky.internal._ATProtocol
import work.socialhub.kbsky.stream.ATProtocolStream

class _ATProtocolStream(
    apiUri: String,
    streamUri: String,
) : ATProtocolStream {

    protected val atproto: ATProtocol = _ATProtocol(apiUri)

    protected val sync: SyncResource = _SyncResource(this.atproto, streamUri)

    /**
     * {@inheritDoc}
     */
    override fun sync() = sync
}
